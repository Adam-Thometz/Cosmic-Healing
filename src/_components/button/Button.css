.Button {
  font-family: 'Open Sans', 'Helvetica Neue', sans-serif;
  font-size: 16px;
  font-weight: 500;
  border: none;
}

.Button-arrow {
  margin: 8px 0 8px 8px;
}

.primary {
  border-radius: 8px;
  background-color: #B279F1;
  color: #FFFFFF;
  padding: 8px 16px;
  box-shadow: 0px 4px 35px 0px #782DCA26;
}

.secondary {
  padding: 8px 24px;
  background-color: #000;
  color: #FFF;
  border-radius: 4px;
}

.tertiary {
  display: flex;
  align-items: center;
  background-color: rgba(0,0,0,0);
  border: 2px solid black;
  border-radius: 5px;
  padding: 4px 16px;
}