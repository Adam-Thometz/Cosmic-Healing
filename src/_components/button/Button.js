import React from 'react';

const Button = ({  primary = false,  secondary = false, children }) => {
  const className = `Button ${primary ? 'primary' : (secondary ? 'secondary' : 'tertiery')}`;
  return <button className={className}>{children}</button>;
};

export default Button;