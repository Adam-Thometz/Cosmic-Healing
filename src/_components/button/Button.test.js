import React from "react";
import { render } from "@testing-library/react";
import Button from './Button';

describe('Button component', () => {
  it('renders without crashing', () => {
    render(<Button>Hello, I'm a button</Button>);
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<Button>Hello, I'm a button</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});