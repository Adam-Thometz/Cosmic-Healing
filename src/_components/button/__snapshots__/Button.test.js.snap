// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component matches the snapshot 1`] = `
<DocumentFragment>
  <button
    class="Button tertiary"
  >
    Hello, I'm a button
    <img
      alt=""
      class="Button-arrow"
      src="arrow.png"
    />
  </button>
</DocumentFragment>
`;
